### ISSUE TITLE
Change stabilization parameter, tau, in ScalarAdvectionArtDiff

### ISSUE THREAD

**Issue by smpark7** [arfc/moltres#140]
This issue can be closed when the formulation for stabilization parameter, ```tau```, in ```ScalarAdvectionArtDiff``` follows the artificial diffusion scheme. A derivation is provided in this [paper](http://www.cimne.com/eoOld/publicaciones/files/PI183.pdf) under Section 3.1.

The formula for tau is given as:
![image](https://user-images.githubusercontent.com/30237775/98233706-64b5e900-1f9a-11eb-840c-91194c8c7716.png)

**Comment by robfairh** [comment]
@smpark7 I have approved #144 because the implementation is correct.
Could you add more detail to this issue about why this is important for Moltres?
I'll merge the PR once you have replied to this question (so the PR doesn't close the issue).

**Comment by smpark7** [comment]
The current ```PrecursorAction``` uses DG-FEM with delayed neutron precursor concentrations as constant monomial variables for simulating precursor behavior. While this has been quite adequate and accurate for past test cases (e.g. MSRE, MSFR), I've encountered significant truncation error in high-gradient regions (e.g. near boundaries with high leakage) due to the zeroth order variables.

For the most part, the overall impact on the neutron flux is minor because the truncation error affects a small fraction of the precursors, which form a small fraction of the neutron source. But the truncation error is very evident when we compare the precursor concentrations with results from other MSR simulation software for benchmarking. As a solution, it would be computationally cheaper to initialize the concentrations as first order variables than to refine the mesh when we consider the overall DOF count.

Using first order Lagrange/L2-Lagrange variables requires additional stabilization techniques in high Schmidt number flows (convection-dominant flows). The ```ScalarAdvectionArtDiff``` kernel introduces an artificial diffusion term that helps to stabilize the problem. The previous iteration of ```ScalarAdvectionArtDiff``` had ```tau``` as a user-defined constant. Thus, I've introduced the proper formulation for ```tau``` in accordance with the reference text.
